### PR TITLE
Cleanup of the codebase

### PR DESCRIPTION
**Pull request by MiroslavGatsanoga**
# Description

## What

The purpose of the PR is to increase the cohesion of the code and decrease the coupling introduced by the usage of many global variables. Doing so will enable us to implement new capability monitoring feature in the service (that will be done in a separate PR).

I did some extra  improvements such as:
- Adding the golang-ci orb
- Fixed some failing unit tests
- Removed code that was not being used anymore e.g. etcd config, puppet files
- Fixed a bunch of linter errors along the way

## Why

[JIRA ticket](https://financialtimes.atlassian.net/browse/UPPSF-1851).

## Anything, in particular, you'd like to highlight to reviewers

The code could use more improvements but as the PR consists of 17 commits affecting 60 files already I would appreciate a "sanity check" review, just to be sure that I have not introduced some fault in the service. Another thing that would be helpful will be to let me know if the end result of the project structure is understandable enough.

## Scope and particulars of this PR (Please tick all that apply)

- [X] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [ ] Feature
- [ ] Documentation
- [ ] Breaking change
- [X] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

**Review by epavlova — APPROVED** [review]

File: healthcheck.go [hunk 1]
```diff
@@ -355,3 +357,14 @@ func buildFtHealthcheckUrl(endpoint url.URL, health string) (string, error) {
 func buildAwsHealthcheckUrl(serviceUrl string) (string, error) {
 	return serviceUrl + "healthCheckDummyFile", nil
 }
+
+func cleanupResp(resp *http.Response) {
```
[the remark this answers is not in the document]
Comment (MiroslavGatsanoga) [reply healthcheck.go:369]: Do you mean the `io.Copy(ioutil.Discard, resp.Body)`? I think there was a moment when this was the way to make sure that the connection will be reused (example discussion [here](https://forum.golangbridge.org/t/do-i-need-to-read-the-body-before-close-it/5594/13)). I think that `resp.Body.Close()` will do that now ([link to implementation](https://github.com/golang/go/blob/7307e86afda3c5c7f6158d2469c39606fd1dba65/src/net/http/transfer.go#L970)).
Not 100% sure if we can safely remove `io.Copy(ioutil.Discard, resp.Body)` as there is no benchmark that shows the performance of this part of the codebase.

File: sender/aggregator.go [hunk 2]
```diff
@@ -1,21 +1,23 @@
-package main
+package sender
```
[the remark this answers is not in the document]
Comment (MiroslavGatsanoga) [reply sender/aggregator.go:1]: That makes sense, what do you think of a third option - merging the `sender` package into the `metrics` one?

File: envs/file_environments.go [hunk 3]
```diff
@@ -280,18 +290,18 @@ func parseEnvsIntoMap(envs []Environment, envCredentials []Credentials) []string
 
 	//remove envs that don't exist anymore
 	removedEnvs := make([]string, 0)
-	for envName := range environments.envMap {
+	for envName := range environments.EnvMap {
 		if !isEnvInSlice(envName, envs) {
 			log.Infof("removing environment from monitoring: %v", envName)
-			delete(environments.envMap, envName)
+			delete(environments.EnvMap, envName)
```
[the remark this answers is not in the document]
Comment (MiroslavGatsanoga) [reply envs/file_environments.go:296]: Totally missed this "clever" implementation 🤦‍♂️ .

**Review by mitagg — DISMISSED** [review]

**Review by mitagg — APPROVED** [review]
Not much of a Go expert, but the refactoring good and I would like to see how we are going from here with the monitoring capabilities.

**Review by epavlova — APPROVED** [review]

**Pull request merged**